bin/si-mcp-server/src/tools/schemaUpsert.ts

```diff
@@ -0,0 +1,248 @@
+import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
+import { z } from "zod";
+import { SchemasApi, FuncsApi } from "@systeminit/api-client";
+import { apiConfig, WORKSPACE_ID } from "../si_client.ts";
+import {
+  errorResponse,
+  generateDescription,
+  successResponse,
+  withAnalytics,
+} from "./commonBehavior.ts";
+
+const name = "schema-upsert";
+const title = "Create or edit a schema";
+const description = `
+<description>
+Create a new schema or edit an existing schema.
+</description>
+<usage>
+If the user does not specify a change set, ask the user which change set to use or if a new one should be created. If the user gives a schema name to edit, use the schema-find tool to find the corresponding schemaId. If the user is trying to add code to an existing asset function for a schema, get the current code for that schema and add to it. Before writing any asset function code, reference the documentation to get the syntax right. Always put each prop within an asset function in a const before adding it to the asset. Do not mention schema variants or locking/unlocking to the user.
```

Comment:
I'm not sure that locking/unlocking is even exposed. I presume that last sentence was just to avoid it interpreting an API error back to the user?